README.md

```diff
@@ -20,3 +20,45 @@ script. Running the tests on Windows is not currently supported.
 # Unsupported Syntax
 * [Class allocators](http://dlang.org/class.html#allocators). These are deprecated in D2.
 * [Class deallocators](http://dlang.org/class.html#deallocators). These are deprecated in D2.
+
+# Example
+
+```d
+/+dub.sdl:
+dependency "libdparse" version="~>0.7"
++/
+import dparse.ast;
+import std.stdio;
+
+class TestVisitor : ASTVisitor
+{
+    alias visit = ASTVisitor.visit;
+
+    override void visit(const FunctionDeclaration decl)
+    {
+        decl.name.text.writeln;
```

Comment:
Put a nested function in the example and add `this.accept(decl);` here. It will show a bit the point of the visitor pattern (for example on run.dlang)

Reply:
Good idea! I also added an `indentLevel` to make this even clearer.